### PR TITLE
[release-v1.2] Different cron times for branches for periodics

### PR DESCRIPTION
**Pull request by aliok**
/hold

I need to hold this because I first need to test this change in different branches to see if my cron changes work.

Checked:
- [x] 1.2
- [x] 1.1
- [x] 1.0
- [x] 0.26
- [ ] 0.25 --> script here is outdated. There are many diffs with the `openshift/release` configuration. I won't touch this script but manually update the config

### CODE REVIEW

**Comment by aliok** [comment]
/unhold

All good. The script produces the right things in all branches.

**Comment by aliok** [comment]
cc @pierDipi

File: openshift/ci-operator/generate-ci-config.sh [hunk 1]
```diff
@@ -41,6 +41,20 @@ EOF
...
+  elif [[ "$branch" == "knative-v0.26" ]]; then
+    echo '0 3 * * 1-5'
+  elif [[ "$branch" == "knative-v1.0" ]]; then
+    echo '0 5 * * 1-5'
+  elif [[ "$branch" == "knative-v1.1" ]]; then
+    echo '0 7 * * 1-5'
+  elif [[ "$branch" == "knative-v1.2" ]]; then
+    echo '0 9 * * 1-5'
+  fi
+}
```
Comment (pierDipi) [inline openshift/ci-operator/generate-ci-config.sh:56]: I'm wondering if we can avoid hard-coding these branches here? :thinking:

Comment (aliok) [reply openshift/ci-operator/generate-ci-config.sh:56]: yeah, I know what you mean. Alternative is maybe create a file in each branch that has the Cron expression. Then we can read the file in the script.

I thought about that but this hardcoding is at least making things more central and I don't have to look at 5 different branches for the cron expression when I want to see when jobs are running.

I am ok with both if you have a strong opinion.

Comment (aliok) [reply openshift/ci-operator/generate-ci-config.sh:56]: Or, any other ideas?

Comment (devguyio) [reply openshift/ci-operator/generate-ci-config.sh:56]: I'm fine with starting with this to validate the idea and not spend much effort in crappy shell until it's proven, then move to a more dynamic random distribution using modulo 
https://stackoverflow.com/questions/13104478/uniformity-of-random-numbers-taken-modulo-n

File: openshift/ci-operator/generate-ci-config.sh [hunk 2]
```diff
@@ -115,22 +129,26 @@ EOF
 }
 
 function print_openshift_47_tests {
+  cron=$(generate_cron_expression)
```
Comment (devguyio) [inline openshift/ci-operator/generate-ci-config.sh:132]:
```suggestion
  cron="$(generate_cron_expression)"
```

**Review by devguyio — COMMENTED** [review]
Is this just to test on 1.2? we try to create those changes on main then /cherry-pick back to release branches.

**Comment by aliok** [comment]
> Is this just to test on 1.2? we try to create those changes on main then /cherry-pick back to release branches.

You cannot run the script on `main` and that's why I wanted to create the PR on 1.2, in case somebody wants to test the changes.

**Comment by aliok** [comment]
/close

Closing in favor of https://github.com/openshift/knative-eventing/pull/1639

File: openshift/ci-operator/generate-ci-config.sh (hunk 1 above)
Comment (pierDipi) [reply openshift/ci-operator/generate-ci-config.sh:56]: https://github.com/openshift/knative-eventing/pull/1639#discussion_r807849476

**Pull request closed without merging**